packages/storage/exp/index.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { _registerComponent, registerVersion } from '@firebase/app-exp';
+import { StringFormat } from '../src/implementation/string';
+import { TaskEvent, TaskState } from '../src/implementation/taskenums';
+
+import { XhrIoPool } from '../src/implementation/xhriopool';
+import { Reference } from '../src/reference';
+import { StorageService } from '../src/service';
+import {
+  Component,
+  ComponentType,
+  ComponentContainer
+} from '@firebase/component';
+
+import { name, version } from '../package.json';
+
+export { ref } from '../src/service';
+export {
+  uploadBytes,
+  uploadString,
+  getMetadata,
+  updateMetadata,
+  list,
+  listAll,
+  getDownloadURL,
+  deleteObject
+} from '../src/reference';
```

Comment:
These seem to be missing from the typings file.

Reply:
I think we can generate typings from the source as long as the exported functions use public types defined in `storage-types`.